src/sequential.cpp

```diff
@@ -326,6 +326,24 @@ void Sequential::backward()
                               this->input_state_update);
 }
 
+void Sequential::smoother()
+/*
+ */
+{
+    // Hidden layers
+    for (auto layer = this->layers.begin(); layer != this->layers.end();
+         layer++) {
+        auto *current_layer = layer->get();
+        auto *next_layer = (layer + 1)->get();
+
+        // Smooth only for LSTM layer
+        if (current_layer->get_layer_name() == "LSTM") {
```

Comment:
Do we need to use smoother for Linear as well?

Reply:
yes. We do need. Because we want to smooth the z_output which is the last layer in (LSTM, LSTM, Linear). That can be solved by creating SLSTM and SLinear layers.

Reply:
@van-dai-vuong I will convert your PR to draft. You can convert back to PR when it is ready

Reply:
@lhnguyen102 ok

Reply:
@lhnguyen102 I put the PR back to ready for review